observability/micrometer/src/test/java/org/wildfly/extension/micrometer/registry/ApplicationRegistryTest.java

```diff
@@ -0,0 +1,263 @@
+        appRegistry1.summary("app1_summary");
+
+        appRegistry2 = new ApplicationRegistry(DEPLOYMENT2, systemRegistry);
+        var app2Counter = appRegistry2.counter(METER_APP);
+        app2Counter.increment();
+        appRegistry2.timer("app2_timer");
+        appRegistry2.gauge("app2_gauge", new ArrayList<String>(), List::size);
+        appRegistry2.summary("app2_summary");
+    }
+
+    @Test
+    public void verifySystemMeters() {
+        Assert.assertNotNull(systemRegistry.find(METER_SYSTEM).meter());
+        Assert.assertNotNull(systemRegistry.find(METER_JVM_MEMORY).meter());
+    }
+
+    @Test
+    public void verifySystemMeterVisibility() {
+        Assert.assertFalse(containsWrongDeploymentTag(appRegistry1.counter(METER_SYSTEM).getId(), DEPLOYMENT1));
+        Assert.assertFalse(containsWrongDeploymentTag(appRegistry2.counter(METER_SYSTEM).getId(), DEPLOYMENT2));
+    }
+
+    @Test
+    public void verifyAppMeterIsolation() {
+        Counter counter1 = appRegistry1.counter(METER_SYSTEM);
+        Counter counter2 = appRegistry2.counter(METER_SYSTEM);
+
+        Counter counter3 = appRegistry1.counter(METER_APP);
+        Counter counter4 = appRegistry2.counter(METER_APP);
+
+        // System metrics should not be visible
+        Assert.assertEquals(0d, counter1.count(), 0.0);
+        assertMetricBelongsToDeployment(counter1.getId(), DEPLOYMENT1);
+        Assert.assertEquals(0d, counter2.count(), 0.0);
+
+        Assert.assertEquals(5d, counter3.count(), 0.0);
+        assertMetricBelongsToDeployment(counter3.getId(), DEPLOYMENT1);
+        Assert.assertEquals(1d, counter4.count(), 0.0);
+        assertMetricBelongsToDeployment(counter4.getId(), DEPLOYMENT2);
+    }
+
+    @Test
+    public void testFind() {
+        // Should find meters from correct registry
+        Assert.assertNotNull(appRegistry1.find(METER_APP).tag(TAG_WF_DEPLOYMENT, DEPLOYMENT1).counter());
+        Assert.assertNotNull(appRegistry2.find(METER_APP).tag(TAG_WF_DEPLOYMENT, DEPLOYMENT2).counter());
+        // Should NOT find meters from other registry
+        Assert.assertNull(appRegistry1.find(METER_APP).tag(TAG_WF_DEPLOYMENT, DEPLOYMENT2).counter());
+        Assert.assertNull(appRegistry2.find(METER_APP).tag(TAG_WF_DEPLOYMENT, DEPLOYMENT1).counter());
+    }
+
+    @Test
+    public void testFindForSystemMetrics() {
+        // System metrics should not be visible
+        Assert.assertNull(appRegistry1.find(METER_JVM_MEMORY).meter());
+        Assert.assertNull(appRegistry2.find(METER_JVM_MEMORY).meter());
+    }
+
+    @Test
+    public void testGetForSystemMetrics() {
+        try {
+            appRegistry1.get(METER_SYSTEM).meter();
+            appRegistry1.get(METER_JVM_MEMORY).meter();
+            appRegistry2.get(METER_SYSTEM).meter();
+            appRegistry2.get(METER_JVM_MEMORY).meter();
+            Assert.fail("System metrics should not be visible.");
```

Comment:
L128-130 could all not fail but L131 throws an exception, and this test would incorrectly pass.